src/ophyd_async/epics/areadetector/drivers/ad_base.py

```diff
@@ -1,13 +1,32 @@
 import asyncio
-from typing import Sequence
+from enum import Enum
+from typing import Sequence, Set
 
-from ophyd_async.core import ShapeProvider
+from ophyd_async.core import AsyncStatus, ShapeProvider
+from ophyd_async.core.signal import set_and_wait_for_value
 
 from ...signal.signal import epics_signal_rw
 from ..utils import ImageMode, ad_r, ad_rw
 from ..writers.nd_plugin import NDArrayBase
 
 
+class DetectorState(str, Enum):
+    Idle = "Idle"
+    Acquire = "Acquire"
+    Readout = "Readout"
+    Correct = "Correct"
+    Saving = "Saving"
+    Aborting = "Aborting"
+    Error = "Error"
+    Waiting = "Waiting"
+    Initializing = "Initializing"
+    Disconnected = "Disconnected"
+    Aborted = "Aborted"
```

Comment:
we've been bitten before by making enums for things and then realising that epics changes enum members at runtime; how comprehensively have these enums been checked? Otherwise on line 40 where you define the detector_state property on ADBase I can imagine some devices will just fail to instantiate because, e.g. the epics screen actually is missing some of these states (or some IOCs have custom ones...)

Reply:
Again after speaking with @coretl seems like these enums are pretty standard. If we come across issues we can always re-define the enum and the DetectorState definition by subclassing ADBase.